app/domain/authentication/authn_gcp/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,88 @@
+require 'command_class'
+
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of restrictions extraction that are set on a Conjur host or user as annotations.
+    ExtractResourceRestrictions = CommandClass.new(
```

Comment:
this class can also be implemented under the `Authentication` module as it's not `authn-gcp` specific. Do we want to implement it that way or start like this and have a leftover to extract it (among other places)?

Reply:
the other option :) since we may need some modifications to use it in other places

Reply:
np